### ISSUE TITLE
Show post ID in postselect dropdown option labels

### ISSUE THREAD

**Issue by danieliser** [PopupMaker/Popup-Maker#660]
`(1): Hello World`

Or

`Hello World <small>(1)</small>`

So that if you have 2 pages with the same name/slug you can tell the difference.

**Comment by fpcorso** [comment]
@danieliser Just to clarify: Which dropdown is this referring to? The selected page/post conditions in the "Targeting" section of the popup editor?

**Comment by danieliser** [comment]
@fpcorso Yea, plenty of sites with multiple pages using same name, hard to differentiate them in those dropdowns.

**Comment by fpcorso** [comment]
@danieliser The conditions system is a web of files and scripts. After A LOT of hunting, narrowed down to somewhere within pumselect2. We really need to add some sort of documentation/wiki for these systems as I still am not sure how this all works even after spending an hour going through all these files.

First, the _selected conditions are defined here as postselect type: https://github.com/PopupMaker/Popup-Maker/blob/master/classes/Conditions.php#L235

Which gets called to render in conditions.js: https://github.com/PopupMaker/Popup-Maker/blob/master/assets/js/src/admin/popup-editor/plugins/conditions.js#L124

Which calls field in templates.js: https://github.com/PopupMaker/Popup-Maker/blob/master/assets/js/src/admin/general/plugins/templates.js#L384

After a lot of debugging, I found that, upon clicking, it calls the AJAX and processResults in select2.js: https://github.com/PopupMaker/Popup-Maker/blob/master/assets/js/src/admin/general/plugins/select2.js#L30

Which queries the DB in object_search method in Ajax.php: https://github.com/PopupMaker/Popup-Maker/blob/master/classes/Admin/Ajax.php#L19

This eventually fires of a prototype.option of the pumselect2 in the vendor folder: https://github.com/PopupMaker/Popup-Maker/blob/master/assets/js/src/admin/general/vendor/select2.full.custom.js#L3148


I am unsure where we would want to modify the option labels for the dropdown. Should we modify in select2.js in the processResults to foreach over all items and change text to include the ID? Should we modify the Ajax object_search to include the ID in the `text` key? Both of those seems to be used a variety of places so not sure what that would affect.

**Comment by danieliser** [comment]
@fpcorso My first and current thought is modifying the AJAX object search method to simply change the labels returned to include a prefix/suffix of the post ID.

- [ ] https://github.com/PopupMaker/Popup-Maker/blob/master/classes/Admin/Ajax.php#L43-L48
- [ ] https://github.com/PopupMaker/Popup-Maker/blob/master/classes/Admin/Ajax.php#L60-L65

While we are at it, might as well support it for taxonomies and users as well here:

- [ ] https://github.com/PopupMaker/Popup-Maker/blob/master/classes/Admin/Ajax.php#L79-L84
- [ ] https://github.com/PopupMaker/Popup-Maker/blob/master/classes/Admin/Ajax.php#L96-L101
- [ ] https://github.com/PopupMaker/Popup-Maker/blob/master/classes/Admin/Ajax.php#L115-L120
- [ ] https://github.com/PopupMaker/Popup-Maker/blob/master/classes/Admin/Ajax.php#L133-L138

**Comment by fpcorso** [comment]
@danieliser Then we don't re-use the select2 pinging that AJAX endpoint anywhere else besides in the conditions, right?

Okay, I'll make those changes and get them committed.

**Comment by fpcorso** [comment]
@danieliser Done. I originally did just the number `(15)` but it wasn't super clear what that number was so I did `(ID: 15)` instead.
![image](https://user-images.githubusercontent.com/9730040/89551133-94f1be00-d7d8-11ea-8878-c7e0b112d65e.png)

**Comment by danieliser** [comment]
@fpcorso that is perfect, was exactly what I had envisioned. That solves at least 5-10 tickets a year if not a lot more as that is only what I see. Also just more polished now.
